sdk/iot/core/tests/cmocka/az_iot_sas_token_tests.c

```diff
@@ -19,7 +22,38 @@
 #define TEST_EXPIRATION_STR "1578941692"
 #define TEST_KEY_NAME "iothubowner"
 
-/*
+#ifndef NO_PRECONDITION_CHECKING
+
+// This block defines the resources needed to verify precondition checking.
+// The macro assert_precondition_checked(func) alone shall be used to assert if a precondition is verified.
+// If a precondition is not verified within a function the result will be either:
+// - A crash in the test, as the the target function will continue to execute with an invalid argument, or
+// - An assert failure indicating a precondition was not tested.
+// Notice that:
+// - If assert_precondition_checked(func) is used, the module must include <setjmp.h>;
+// - If a function has two precondition checks and both are supposed to fail on a given test, assert_precondition_checked(func)
+//   is unable to distinguish which precondition has failed first. Testing precondition checking separately is advised.
+// - Tests using assert_precondition_checked(func) must not be run in parallel (!).
+
+static jmp_buf g_jmp_buf; 
+static unsigned int precondition_test_count = 0;
+static void az_precondition_test_failed_fn()
```

Comment:
Is there any way we can have this code in a .c or headers so it doesn't have to be copied over for each new test file?

Reply:
Makes sense. @JeffreyRichter , do you think this solution is good enough to go into the core SDK? And if yes, where would you recommend it to go?

Reply:
Not in the SDK but it can go here in this folder: https://github.com/Azure/azure-sdk-for-c/tree/master/sdk/core/core/test/cmocka

Reply:
Moved these macros into az_precondition_internal.h

Reply:
No, az_precondition_internal.h is part of the SDK. and not in the folder I suggested.